### PR TITLE
ADDED: player creation depending on clients

### PR DESCRIPTION
**Pull request by MauxC**

### CODE REVIEW

**Review by Varriss — DISMISSED** [review]

File: src/IScene.hpp [hunk 1]
```diff
@@ -45,15 +48,15 @@ namespace ecs
          * @brief Set the callback function to call when an entity is added
          * @param callback Callback function
          */
-        virtual void setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>)> callback) = 0;
+        virtual void setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>, SceneType)> callback) = 0;
```
Comment (iDommel) [inline src/IScene.hpp:51]: Missing brief

File: src/Scene.cpp [hunk 2]
```diff
@@ -14,7 +14,7 @@
 namespace ecs
 {
 
-    Scene::Scene(std::function<std::unique_ptr<IScene>()> init) : _initFunc(init)
+    Scene::Scene(std::function<std::unique_ptr<IScene>()> init, SceneType scene) : _initFunc(init), _type(scene)
```
Comment (iDommel) [inline src/Scene.cpp:17]: Missing brief

File: src/Scene.hpp [hunk 3]
```diff
@@ -28,7 +29,7 @@ namespace ecs
          * @brief Scene constructor
          * @param init Scene init function
          */
-        Scene(std::function<std::unique_ptr<IScene>()> init);
+        Scene(std::function<std::unique_ptr<IScene>()> init, SceneType type);
```
Comment (iDommel) [inline src/Scene.hpp:32]: Missing brief

File: src/Scene.hpp [hunk 4]
```diff
@@ -54,15 +55,15 @@ namespace ecs
          * @brief Set the callback function to call when an entity is added
          * @param callback Callback function
          */
-        void setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>)> callback);
+        void setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>, SceneType)> callback);
```
Comment (iDommel) [inline src/Scene.hpp:58]: Missing brief

File: src/SceneManager.cpp [hunk 5]
```diff
@@ -73,12 +73,12 @@ namespace ecs
...
     }
 
-    std::map<SceneManager::SceneType, std::unique_ptr<IScene>> &SceneManager::getScenes()
+    std::map<SceneType, std::unique_ptr<IScene>> &SceneManager::getScenes()
     {
         return _scenes;
     }
 
-    void SceneManager::setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>)> callback)
+    void SceneManager::setAddEntityCallback(std::function<void(std::shared_ptr<IEntity>, SceneType)> callback)
```
Comment (iDommel) [inline src/SceneManager.cpp:81]: Missing brief

File: src/systems/ANetworkSystem.hpp [hunk 6]
```diff
@@ -46,7 +46,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity);
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene);
```
Comment (iDommel) [inline src/systems/ANetworkSystem.hpp:49]: Missing brief

File: src/systems/AudioSystem.hpp [hunk 7]
```diff
@@ -27,7 +27,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/AudioSystem.hpp:30]: Missing brief

File: src/systems/CollideSystem.hpp [hunk 8]
```diff
@@ -35,7 +35,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/CollideSystem.hpp:38]: Missing brief

File: src/systems/EventSystem.hpp [hunk 9]
```diff
@@ -32,14 +32,14 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/EventSystem.hpp:35]: Missing brief

File: src/systems/GameSystem.hpp [hunk 10]
```diff
@@ -56,7 +56,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/GameSystem.hpp:59]: Missing brief

File: src/systems/GraphicSystem.cpp [hunk 11]
```diff
@@ -92,7 +92,7 @@ namespace ecs
         std::cerr << "GraphicSystem::destroy" << std::endl;
     }
 
-    void GraphicSystem::onEntityAdded(std::shared_ptr<IEntity> entity)
+    void GraphicSystem::onEntityAdded(std::shared_ptr<IEntity> entity, SceneType)
```
Comment (iDommel) [inline src/systems/GraphicSystem.cpp:95]: Missing brief

File: src/systems/GraphicSystem.hpp [hunk 12]
```diff
@@ -31,7 +31,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/GraphicSystem.hpp:34]: Missing brief

File: src/systems/ISystem.hpp [hunk 13]
```diff
@@ -39,7 +40,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        virtual void onEntityAdded(std::shared_ptr<IEntity> entity) = 0;
+        virtual void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) = 0;
```
Comment (iDommel) [inline src/systems/ISystem.hpp:43]: Missing brief

File: src/systems/ParticlesSystem.hpp [hunk 14]
```diff
@@ -25,7 +25,7 @@ namespace ecs
          * @brief The callback to be called when an entity is added to a scene
          * @param entity The Entity that was added
          */
-        void onEntityAdded(std::shared_ptr<IEntity> entity) final;
+        void onEntityAdded(std::shared_ptr<IEntity> entity, SceneType scene) final;
```
Comment (iDommel) [inline src/systems/ParticlesSystem.hpp:28]: Missing brief

**Review by iDommel — CHANGES_REQUESTED** [review]
Briefs

File: src/IScene.hpp [hunk 15]
```diff
@@ -45,15 +48,15 @@ namespace ecs
          * @brief Set the callback function to call when an entity is added
          * @param callback Callback function
          */
```
Comment (Varriss) [inline src/IScene.hpp:50]: scene type param missing in doxygen

File: src/systems/ANetworkSystem.cpp [hunk 16]
```diff
@@ -21,172 +21,14 @@ namespace ecs
 
     ANetworkSystem::ANetworkSystem() : QObject(nullptr)
     {
-        _serverAddr = QHostAddress::LocalHost;
+        _serverAddr = QHostAddress(QStringLiteral("10.42.0.1"));
```
Comment (Varriss) [inline src/systems/ANetworkSystem.cpp:24]: Should we make a Define for the host address ?

File: src/systems/NetworkServerSystem.hpp [hunk 17]
```diff
@@ -35,27 +36,31 @@ namespace ecs {
         /// @param msg Message to send
         /// @param clientId Client ID to send msg to
         void writeToClient(const std::string &msg, int clientId);
+        void writeToClient(const std::string &msg, std::pair<QString /*addr*/, unsigned short /*port*/> client);
```
Comment (Varriss) [inline src/systems/NetworkServerSystem.hpp:39]: comments inside function declaration

**Review by Varriss — DISMISSED** [review]

File: src/IScene.hpp (hunk 15 above)
Comment (MauxC) [reply src/IScene.hpp:50]: It's wrong !!

Comment (MauxC) [reply src/IScene.hpp:50]: It's wrong !!

**Pull request merged**